runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -1,12 +1,196 @@
+                    // the following scenarios:
+                    // 1. Request to subscribe is received before the Log Replication JVM or pod starts and
+                    // initializes the table
+                    // 2. Topology change which removes this cluster from the status table.
+                    // We cannot differentiate between the two cases here so continue with the right
+                    // behavior for the 1st case, i.e., subscribe and wait for the initial Snapshot Sync to get
+                    // triggered.  If we are here because of the second case, the listener will not get any updates
+                    // from LR and subscription will be a no-op.
+                    if (entries.isEmpty()) {
+                        log.info("No record for the Logical Group Model found in the Status Table.  Subscription " +
+                                "could have been attempted before LR startup.  Subscribe the listener and wait for" +
+                                "initial Snapshot Sync");
+                    } else {
+                        // Sharding in Log Replication is done based on a destination.  So for a given replication model,
+                        // any Sink node will have a session with only 1 Source node.  Hence, there should be only 1
+                        // record corresponding to the Logical Group Replication Model.
+                        Preconditions.checkState(entries.size() == 1);
+                        entry = entries.get(0);
+                    }
+
+                    boolean snapshotSyncInProgress = false;
+
+                    if (entry == null || entry.getPayload().getSinkStatus().getDataConsistent()) {
+                        // No snapshot sync is in progress
+                        log.info("No Snapshot Sync is in progress.  Request the client to perform a full sync on its " +
+                            "tables.");
+                        Optional<Timer.Sample> clientFullSyncTimer = MicroMeterUtils.startTimer();
+                        clientListener.performFullSync(txnContext);
+                        MicroMeterUtils.time(clientFullSyncTimer, "logreplication.client.fullsync.duration");
+                    } else {
+                        // Snapshot sync is in progress.  Subscribe without performing a full sync on the tables.
+                        log.info("Snapshot Sync is in progress.  Subscribing without performing a full sync on client" +
+                            " tables.");
+                        snapshotSyncInProgress = true;
+                    }
+                    txnContext.commit();
+
+                    // Subscribe from the snapshot timestamp of this transaction, i.e., log tail when the transaction started.
+                    // Subscribing from the commit address will result in missed updates which took place between the start
```

Comment:
Regarding this statement if its a read-only tx both should match, isn't it?

Reply:
Ok, agree. It was not clear from the method description so was not sure it was happening.